### ISSUE TITLE
Joi schema validation in a test

### ISSUE THREAD

**Issue by Jakems777** [vlucas/frisby#478]
Has anyone tried using Joi schema validate approach in Frisby tests?
Any examples?

**Comment by H1Gdev** [comment]
@Jakems777 

**Examples**

[jsonTypes](https://www.frisbyjs.com/expectations-assertions.html)

[expects_jsonTypes_spec.js](https://github.com/vlucas/frisby/blob/master/__tests__/expects_jsonTypes_spec.js)

**Comment by Jakems777** [comment]
@H1Gdev Thank you for your response and the examples, that will come in handy!

What I was going after was something more like this:
https://qiita.com/H1Gdev/items/7aa5ca8a78839c615ac4
this example (seems its you there as well :D ) has this code: 

const frisby = require('frisby');
const Validator = require('jsonschema').Validator; // install jsonschema.

beforeAll(function () {
  frisby.addExpectHandler('jsonschema', function (response, schema) {
    var validator = new Validator();
    validator.validate(response.json, schema, { throwError: true });
  });
});

afterAll(function () {
  // 必要であれば削除する
  frisby.removeExpectHandler('jsonschema');
});

describe('Custom Expect Handler', function () {
  it('Test', function (doneFn) {
    frisby.fromJSON(4)
      .expect('jsonschema', { 'type': 'number'})
      .done(doneFn);
  });
});

and by Joi schema validate, I meant something like this: 
https://scotch.io/tutorials/node-api-schema-validation-with-joi
so Imagine something like this in your tests:

const schema = Joi.object().keys({  
        email: Joi.string().email().required(),     
        phone: Joi.string().regex(/^\d{3}-\d{3}-\d{4}$/).required(),    
        birthday: Joi.date().max('1-1-2004').iso(),
    });

const frisby = require('frisby');
const Joi = frisby.Joi; // Frisby exports Joi for convenience on type assersions
const conf = require("../../conf.js");


describe('Random test', function () {
    it('get something get status 200', function (done) {
        return frisby
            .get("whatever endpoint")
            .expect('status', 200)
            .expect('jsonTypes',  {
                schema
            });
    });
});
or after the get or post something custom like this: 

    // validate the request data against the schema
    Joi.validate(data, schema, (err, value) => {
        if (err) {
       //  fail the IT block from frisby/jasmine
            });
        } else {
         done()
        }
    });
});

So, basically I though that there is a way to define schema/schemas for some tests and just use a Joi or jasmine "expect" on it, without having to chain a lot of different expects. (the schema comes in handy only if working with a huge JSON payload . . .)

**Comment by vlucas** [comment]
Something not covered well in the docs is that you can use the `use` function to better abstract and decouple things for code re-use across your tests.

Example:
```javascript
// Define a new function
function validateBlogPost(spec) {
    return spec.expect('jsonTypes', 'items.*', { // Assert *each* object in 'items' array
      'id': Joi.string().required(),
      'url': Joi.string().uri().required(),
      'title': Joi.string().required(),
      'date_published': Joi.date().iso().required(),
    });
}

// Now 'use' it in your test:
describe('Random test', function () {
    it('get something get status 200', function (done) {
        return frisby
            .get("whatever endpoint")
            .expect('status', 200)
            .use(validateBlogPost);
    });
});
```

**Comment by Jakems777** [comment]
@vlucas Thats cool, thank you!
